pool/acceptedwork.go

```diff
@@ -45,70 +45,93 @@ func bigEndianBytesToHeight(b []byte) uint32 {
 }
 
 // AcceptedWorkID generates a unique id for work accepted by the network.
-func AcceptedWorkID(blockHash string, blockHeight uint32) []byte {
+func AcceptedWorkID(blockHash string, blockHeight uint32) ([]byte, error) {
```

Comment:
I prefer writing the components of the id, I'll revert the error handling being done for each `Write` or `WriteString`. I initially wanted to cover all possible error avenues with the updated/defined error types. I realize that is not necessary here however.

Reply:
If we are using funcs which can return errors, we should really be handling those errors, even if we dont expect them to occur. Not handling these errors can introduce bugs which are not noticed, or are very difficult to track down.

At the very least, I would like to see the errors caught with a blank identifier, so it is obvious to anybody reading the code that an error is being ignored here. For example...

```
_, _ = buf.WriteString(hex.EncodeToString(heightToBigEndianBytes(blockHeight)))
```

Reply:
Noted, will indicate the errors are being ignored where with blank identifiers, should help with readability as well. The reason why its ok to ignore the errors related to the buffer is that the only time the buffer will error is if there is an out of memory error. When that happens dcrpool will panic and the added error handling will actually not report details expected because of the panic. It is ok to ommit error handling for buffers here as a result.

Reply:
Makes sense. Thanks for providing the extra info and adding the blank identifiers.